lib/sky-pages-module-generator.js

```diff
@@ -15,6 +17,31 @@ const routeGenerator = require('./sky-pages-route-generator');
  */
 function getSource(skyAppConfig) {
 
+  // Generate style `require()` statements.
+  const cssFilePaths = skyAppConfig.skyux.app &&
+    skyAppConfig.skyux.app.styles &&
+    skyAppConfig.skyux.app.styles;
+
+  let styles = '';
+  if (cssFilePaths) {
+    styles = cssFilePaths
+      .filter(path => {
```

Comment:
No doubt, for most use-cases this array will be very small.  I definitely like how readable the `map/filter/reduce` pattern is, I'm worried about the performance should the array become large.  In this case, I think the code could be just as readable using a single for loop.  Something like:

```
let styles = '';
let hasExternal = false;

if (cssFilePaths) {
  for (const i = 0, const len = cssFilePaths.length; i < len; i++) {
    if (/^http/.test(cssFilePaths[i])) {
      hasExternal = true;
    } else {
      styles += `require('style-loader!${path}');\n`);
    }    
  }
}

If (hasExternal) {
  logger.warn(...);
}
```

This has a few benefits without sacrificing code readability:

- the warning is only shown once despite the number of externals listed (which you could of course still solve using the other pattern).
- Array is only looped through once

I'm happy to discuss further if you disagree @Blackbaud-SteveBrush.